tasks/first_server.yml

```diff
@@ -127,3 +127,22 @@
   ansible.builtin.set_fact:
     active_server: "{{ inventory_hostname }}"
   run_once: true
+
+- name: Get all nodes
+  ansible.builtin.shell: |
+    {{ rke2_data_path }}/bin/kubectl --kubeconfig /etc/rancher/rke2/rke2.yaml \
+    get nodes --no-headers | awk '{print $1}'
+  args:
+    executable: /bin/bash
+  changed_when: false
+  register: node_names
+
+- name: Restore etcd - remove old nodes
+  ansible.builtin.shell: |
+    {{ rke2_data_path }}/bin/kubectl --kubeconfig /etc/rancher/rke2/rke2.yaml \
+    delete node {{ item }} 2>&1 || true
+  args:
+    executable: /bin/bash
+  with_items: "{{ node_names.stdout_lines | difference(groups[rke2_cluster_group_name]) }}"
+  changed_when: false
+  when: inventory_hostname != item and (do_etcd_restore is defined or do_etcd_restore_from_s3 is defined)
```

Comment:
Please correct the yaml syntax and add an empty line to the file.

Reply:
Added an new line to the end of the file

Reply:
thx